File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */
var gcdOfStrings = function(str1, str2) {
    if(str1 + str2 !== str2 + str1) return "";
    
    let word1 = str1.length;
    let word2 = str2.length;
    
    function gcd(a, b) {
        if(b === 0) return a;
        let temp = a % b;
        return gcd(b, temp);
    }
    
    return str1.slice(0, gcd(word1, word2));
};

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        def gcd(a, b):
            while b:
                a, b = b, a % b
            return a
        
        if (str1 + str2) != (str2 + str1):
            return ""
        
        s1 = len(str1)
        s2 = len(str2)
        x = gcd(s1, s2)
        return str1[:x]
        